src/ape/api/networks.py

```diff
@@ -204,21 +204,73 @@ def set_default_network(self, network_name: str):
     def encode_deployment(
         self, deployment_bytecode: HexBytes, abi: ConstructorABI, *args, **kwargs
     ) -> "TransactionAPI":
-        ...
+        """
+        Create a deploy transaction.
+
+        Args:
+            deployment_bytecode (HexBytes): The bytecode to deploy.
+            abi (ConstructorABI): The constructor interface of the contract.
+            *args: Constructor arguments.
+            **kwargs: Transaction arguments.
+
+        Returns:
+            class:`~ape.api.transactions.TransactionAPI`
+        """
 
     @abstractmethod
     def encode_transaction(
         self, address: AddressType, abi: MethodABI, *args, **kwargs
     ) -> "TransactionAPI":
-        ...
+        """
+        Create a transaction object from a contract function call.
+
+        Args:
+            address (AddressType): The address of the contract.
+            abi (MethodABI): The function to call on the contract.
+            *args: Function arguments.
+            **kwargs: Transaction arguments.
+
+        Returns:
+            class:`~ape.api.transactions.TransactionAPI`
+        """
 
     @abstractmethod
     def decode_logs(self, abi: EventABI, raw_logs: List[Dict]) -> Iterator[ContractLog]:
-        ...
+        """
+        Get logs from raw chain data.
+
+        Args:
+            abi (EventABI): The event producing the logs.
+            raw_logs (List[Dict]): A list of raw log data from the chain.
+
+        Returns:
+            Iterator[:class:`~ape.types.ContractLog`]
+        """
 
     @abstractmethod
     def create_transaction(self, **kwargs) -> "TransactionAPI":
-        ...
+        """
+        Create a transaction using key-value arguments.
+
+        Args:
+            **kwargs: Everything the transaction needs initialize.
+
+        Returns:
+            class:`~ape.api.transactions.TransactionAPI`
+        """
+
+    @abstractmethod
+    def decode_calldata(self, abi: MethodABI, raw_data: bytes) -> Tuple[Any, ...]:
```

Comment:
Is this why there's a `type: ignore` here?

https://github.com/ApeWorX/ape/blob/main/src/ape/contracts/base.py#L91